### DISCUSSION TITLE
Support for any cloud URL to upload videos and not just YouTube URL

### DISCUSSION THREAD

**Discussion by manjirisulekavadi** [Sunbird-Ed/Community#575] [Enhancement request]
Has any one supported a requirement to upload video content from other cloud storage URLs like drives and not just from the youtube?  
@Krishnaj20

**Comment by Krishnaj20** [comment]
@manjirisulekavadi - Can you let us know what errors are you facing when uploading these content from these urls?

@ashokreddy1208 - can you check this thread and let us know

**Comment by ashokreddy1208** [comment]
@manjirisulekavadi @Krishnaj20  Currently uploading video content from other cloud storage URLs (Google drive) is not possible, looks like we need to whitelist domain url to make it work, we will check and get back by tomorrow.

cc @amitpriyadarshi

**Reply by manjirisulekavadi** [reply]
Thanks @ashokreddy1208 shall await for the update. 
@manojlondhe

**Reply by Krishnaj20** [reply]
@pallakartheekreddy - Can you check this thread and respond accordingly? Requestor is referring to the upload editor.

**Comment by pallakartheekreddy** [comment]
@manjirisulekavadi Currently, we only support YouTube URLs. In order to enable support for other cloud URLs, some backend & frontend modifications are required. We will be happy to help you to extend Sunbird to build and contribute this change.
@Krishnaj20

**Reply by kameshbhr** [reply]
@pallakartheekreddy Can you please elaborate what you mean only support Youtube URL. Is this a limitation in editor and player or is it also in the create API? Will someone be able to use the content service to create and publish content with any URL - with some configuration?

**Reply by pallakartheekreddy** [reply]
@kameshbhr Currently, we are not allowing any cloud URLs due to concerns about the expiration of those URLs. We encountered an issue when attempting to use a Google Drive URL and as a result, the content associated with that URL became invalid. To ensure the reliability and consistency of content, we have chosen to restrict the usage of cloud URLs in our system.

This limitation exists on both the API side and the front end. Enabling this feature would require making minimal code changes in both areas, rather than just a simple configuration modification.
@maheshkumargangula @vrayulu
